### PR TITLE
gene effect implementation and t tests

### PR DESCRIPTION
**Pull request by Ecohen1763**

### CODE REVIEW

**Comment by Ecohen1763** [comment]
`CRISPRGeneEffect.csv` has been implemented in place of `CRISPRGeneDependency.csv`, the training model is exposed to curated set of genes, and t tests were added post vae

File: 0.data-download/construct_gene_dictionary.ipynb [hunk 1]
```diff
@@ -853,9 +856,9 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python [conda env:gene_dependency_representations] *",
+   "display_name": "base",
    "language": "python",
-   "name": "conda-env-gene_dependency_representations-py"
+   "name": "python3"
```
Comment (gwaybio) [inline 0.data-download/construct_gene_dictionary.ipynb:862]: Please make sure you're running the jupyter notebooks in the correct conda environment

File: 0.data-download/data_downloader.ipynb [hunk 2]
```diff
@@ -100,9 +100,9 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python [conda env:gene_dependency_representations] *",
+   "display_name": "base",
    "language": "python",
-   "name": "conda-env-gene_dependency_representations-py"
+   "name": "python3"
```
Comment (gwaybio) [inline 0.data-download/data_downloader.ipynb:105]: same goes for here too. We should be using the `gene_dependency_representations` kernel

File: 0.data-download/scripts/data_loader.py [hunk 3]
```diff
@@ -13,52 +13,51 @@ def load_data(data_directory, adult_or_pediatric="all", id_column="ModelID"):
...
 
     # Load data
-    dependency_df = (
-        pd.read_csv(dependency_data_file, index_col=0).reset_index().dropna(axis=1)
-    )
     model_df = pd.read_csv(model_file)
- 
+    effect_df = (
+        pd.read_csv(effect_data_file, index_col=0).reset_index().dropna(axis=1)
+    )
```
Comment (gwaybio) [inline 0.data-download/scripts/data_loader.py]:
```suggestion
    effect_df = pd.read_csv(effect_data_file).dropna(axis=1)
```

I think this should effectively do the same thing?

File: 1.data-exploration/scripts/nbconverted/2.test-train_split.py [hunk 4]
```diff
@@ -47,16 +48,35 @@
...
 )
-presplit_dependency_df
+presplit_effect_df
 
 
 # In[6]:
 
 
+# preparing presplit dataframe to be scaled
+col_num = len(presplit_effect_df.columns.to_list())
```
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/2.test-train_split.py]:
```python
col_num = presplit_effect_df.shape[1]
```

Should work. Maybe worth simplifying?

File: 1.data-exploration/scripts/nbconverted/2.test-train_split.py [hunk 5]
```diff
@@ -47,16 +48,35 @@
...
+presplit_effect_scaled_df = presplit_effect_df.iloc[:, 1:col_num-1]
+
+# scaling gene effect data to 0-1 range
+scaler = MinMaxScaler(feature_range=(0,1))
```
Comment (gwaybio) [inline 1.data-exploration/scripts/nbconverted/2.test-train_split.py]: We would prefer scaling to occur separately for training/testing, which would simulate the real world scenario more closely

File: 2.train-VAE/scripts/nbconverted/3.beta_vae_training_extract_model_components.py [hunk 6]
```diff
@@ -55,10 +55,13 @@
...
-largest_var_df = gene_stats.nlargest(1000, "variance")
-gene_list = largest_var_df["gene_ID"].tolist()
-gene_list
+
+# create dataframe containing the genes that passed an initial QC (see Pan et al. 2022) and their corresponding gene label and extract the gene labels
+gene_dict_df = pd.read_csv("../0.data-download/data/CRISPR_gene_dictionary.tsv", delimiter='\t')
+gene_list = []
+for index, row in gene_dict_df.iterrows():
+    if row['qc_pass'] == True:
+        gene_list.append(row['dependency_column'])
```
Comment (gwaybio) [inline 2.train-VAE/scripts/nbconverted/3.beta_vae_training_extract_model_components.py]:
```suggestion
gene_list_passed_qc = gene_dict_df.query("qc_pass").dependency_column.tolist()
```

I think this is the same function, just a bit more pythonic

File: 3.analysis/scripts/nbconverted/3.t_tests.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+ped_ids = []
+adult_ids = []
+male_ids = []
+female_ids = []
+ped_male_ids = []
+adult_male_ids = []
+ped_female_ids = []
+adult_female_ids = []
+
+for index, row in metadata_df.iterrows():
```
Comment (gwaybio) [inline 3.analysis/scripts/nbconverted/3.t_tests.py]: You're going to love [`pandas.DataFrame.query()`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.query.html) 😁

File: 3.analysis/scripts/nbconverted/3.t_tests.py [hunk 8]
```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
```
Comment (gwaybio) [inline 3.analysis/scripts/nbconverted/3.t_tests.py:1]: Please output the results of the t-test.

The deliverable is a 3 `.tsv` file:

- Column 1: t-test statistic
- Column 2: p-value
- Column 3: Latent feature
- Column 4: comparison (e.g. "Male vs. Female")

This tidy data format is called "long". Please add the output file to this repo

**Review by gwaybio — APPROVED** [review]
Looks good! Please address my comments prior to merging

File: 1.data-exploration/scripts/nbconverted/2.test-train_split.py (hunk 5 above)
Comment (Ecohen1763) [reply 1.data-exploration/scripts/nbconverted/2.test-train_split.py]: Sure, implementing now

File: 0.data-download/scripts/data_loader.py (hunk 3 above)
Comment (Ecohen1763) [reply 0.data-download/scripts/data_loader.py]: Yep, adding the edit

File: 1.data-exploration/scripts/nbconverted/2.test-train_split.py (hunk 4 above)
Comment (Ecohen1763) [reply 1.data-exploration/scripts/nbconverted/2.test-train_split.py]: Yeah, much more straightforward

File: 2.train-VAE/scripts/nbconverted/3.beta_vae_training_extract_model_components.py (hunk 6 above)
Comment (Ecohen1763) [reply 2.train-VAE/scripts/nbconverted/3.beta_vae_training_extract_model_components.py]: Yep, adding this simplification

File: 3.analysis/scripts/nbconverted/3.t_tests.py (hunk 7 above)
Comment (Ecohen1763) [reply 3.analysis/scripts/nbconverted/3.t_tests.py]: Working on implementing now

File: 3.analysis/scripts/nbconverted/3.t_tests.py (hunk 8 above)
Comment (gwaybio) [reply 3.analysis/scripts/nbconverted/3.t_tests.py:1]: sorry, I meant "4 column tsv file"

**Comment by Ecohen1763** [comment]
After discussion, we have decided that this pr is ready to merge. Currently `3.beta_vae_training_extract_model_components.ipynb` and `2.gene_set_enrichment_analysis.ipynb` are not running on the `gene_dependency_representations` kernel, but this will be addressed in the near future.

**Pull request merged**